libportability-gfx/src/impls.rs

```diff
@@ -732,69 +756,142 @@ pub extern "C" fn gfxCreateDevice(
-        feat(ef.textureCompressionBC, Features::FORMAT_BC) |
-        feat(ef.occlusionQueryPrecise, Features::PRECISE_OCCLUSION_QUERY) |
-        feat(ef.pipelineStatisticsQuery, Features::PIPELINE_STATISTICS_QUERY) |
-        feat(ef.vertexPipelineStoresAndAtomics, Features::VERTEX_STORES_AND_ATOMICS) |
-        feat(ef.fragmentStoresAndAtomics, Features::FRAGMENT_STORES_AND_ATOMICS) |
-        feat(ef.shaderTessellationAndGeometryPointSize, Features::SHADER_TESSELLATION_AND_GEOMETRY_POINT_SIZE) |
-        feat(ef.shaderImageGatherExtended, Features::SHADER_IMAGE_GATHER_EXTENDED) |
-        feat(ef.shaderStorageImageExtendedFormats, Features::SHADER_STORAGE_IMAGE_EXTENDED_FORMATS) |
-        feat(ef.shaderStorageImageMultisample, Features::SHADER_STORAGE_IMAGE_MULTISAMPLE) |
-        feat(ef.shaderStorageImageReadWithoutFormat, Features::SHADER_STORAGE_IMAGE_READ_WITHOUT_FORMAT) |
-        feat(ef.shaderStorageImageWriteWithoutFormat, Features::SHADER_STORAGE_IMAGE_WRITE_WITHOUT_FORMAT) |
-        feat(ef.shaderUniformBufferArrayDynamicIndexing, Features::SHADER_UNIFORM_BUFFER_ARRAY_DYNAMIC_INDEXING) |
-        feat(ef.shaderSampledImageArrayDynamicIndexing, Features::SHADER_SAMPLED_IMAGE_ARRAY_DYNAMIC_INDEXING) |
-        feat(ef.shaderStorageBufferArrayDynamicIndexing, Features::SHADER_STORAGE_BUFFER_ARRAY_DYNAMIC_INDEXING) |
-        feat(ef.shaderStorageImageArrayDynamicIndexing, Features::SHADER_STORAGE_IMAGE_ARRAY_DYNAMIC_INDEXING) |
-        feat(ef.shaderClipDistance, Features::SHADER_CLIP_DISTANCE) |
-        feat(ef.shaderCullDistance, Features::SHADER_CULL_DISTANCE) |
-        feat(ef.shaderFloat64, Features::SHADER_FLOAT64) |
-        feat(ef.shaderInt64, Features::SHADER_INT64) |
-        feat(ef.shaderInt16, Features::SHADER_INT16) |
-        feat(ef.shaderResourceResidency, Features::SHADER_RESOURCE_RESIDENCY) |
-        feat(ef.shaderResourceMinLod, Features::SHADER_RESOURCE_MIN_LOD) |
-        feat(ef.sparseBinding, Features::SPARSE_BINDING) |
-        feat(ef.sparseResidencyBuffer, Features::SPARSE_RESIDENCY_BUFFER) |
-        feat(ef.sparseResidencyImage2D, Features::SPARSE_RESIDENCY_IMAGE_2D) |
-        feat(ef.sparseResidencyImage3D, Features::SPARSE_RESIDENCY_IMAGE_3D) |
-        feat(ef.sparseResidency2Samples, Features::SPARSE_RESIDENCY_2_SAMPLES) |
-        feat(ef.sparseResidency4Samples, Features::SPARSE_RESIDENCY_4_SAMPLES) |
-        feat(ef.sparseResidency8Samples, Features::SPARSE_RESIDENCY_8_SAMPLES) |
-        feat(ef.sparseResidency16Samples, Features::SPARSE_RESIDENCY_16_SAMPLES) |
-        feat(ef.sparseResidencyAliased, Features::SPARSE_RESIDENCY_ALIASED) |
-        feat(ef.variableMultisampleRate, Features::VARIABLE_MULTISAMPLE_RATE) |
-        feat(ef.inheritedQueries, Features::INHERITED_QUERIES)
+        feat(ef.robustBufferAccess, Features::ROBUST_BUFFER_ACCESS)
+            | feat(ef.fullDrawIndexUint32, Features::FULL_DRAW_INDEX_U32)
```

Comment:
can we configure it so that `| ` is at the end? this would make these lines aligned like they used to be

Reply:
We can use `rustfmt.toml` with
```
binop_separator = "Back"
indent_style = "Visual"
```
which will make all lines look almost the same as origin (no indent and | at the end of the line). Only long lines would be changed:
```
        feat(ef.shaderFloat64, Features::SHADER_FLOAT64) |
        feat(ef.shaderInt64, Features::SHADER_INT64) |
        feat(ef.shaderInt16, Features::SHADER_INT16) |
        feat(ef.shaderResourceResidency,
             Features::SHADER_RESOURCE_RESIDENCY) |
        feat(ef.shaderResourceMinLod, Features::SHADER_RESOURCE_MIN_LOD) |
```
Applying this globally changes a lot of other things as well so I'm personally leaning more towards just wrapping this section in `#[rustfmt::skip] { }`.

Reply:
Well, didn't realize expression attributes are experimental (https://github.com/rust-lang/rust/issues/15701 ). I could probably work around that with macro that just copies its input and tag it with `#[rustfmt::skip::macros()]`, unless there is some other way to achieve this.

Reply:
`indent_style` affects a lot of indentation. It is easier to show in diff: https://github.com/krolli/portability/commit/2ab5d2a37214de2717e1d039b8ad352992d375ec

I am leaning more towards keeping things on default and only disabling it in those places where it makes sense: https://github.com/krolli/portability/commit/6e8e7d15a1265cd39b856409b5b509364c14ad21